engine/src/execution_graph/logic_controllers/CacheMachine.cpp

```diff
@@ -251,11 +258,14 @@ void CacheMachine::addCacheData(std::unique_ptr<ral::cache::CacheData> cache_dat
 	}
 }
 
-void CacheMachine::addToCache(std::unique_ptr<ral::frame::BlazingTable> table, const std::string & message_id) {
-
+void CacheMachine::addToCache(std::unique_ptr<ral::frame::BlazingTable> table, const std::string & message_id, bool always_add) {
 	// we dont want to add empty tables to a cache, unless we have never added anything
-	if (!this->something_added || table->num_rows() > 0){
+	if (!this->something_added || table->num_rows() > 0|| always_add){
 		for (auto col_ind = 0; col_ind < table->num_columns(); col_ind++){
+			if (table->view().column(col_ind).offset() > 0){
```

Comment:
this was one of my changes do we really need it? I am not sure we do. its basically a no op if nothing needs to be done

Reply:
yeah, we are ensuring ownership lower down anyways